Moving Icons away from CC BY-ND 3.0 licensed Icons to help with FOSS Packaging

Docteh opened the pull request:
I've seen some comments stating that sharing pre-compiled binaries
is problematic for linux distributions due to the CC BY-ND 3.0 license

In this commit I am tackling the 16x16 icons

The reported status of the "yuzu Web Service" section of Configuration uses two icons, checked and failed

checked is replaced by material-design-icons check_circle
failed is replaced by material-design-icons error

Colors have been set by taking the white pngs of 18x18, cropping to 16x16, and then adjusting Curves using Paint.NET

Other tools are available.

There is also a locked icon that does not appear to be used in the current code. citra uses it for multiplayer

lock is replaced by material-designs-icons lock

Some of the colorful theme icons are clearly from https://github.com/icons8/flat-color-icons/ which is available MIT licensed

Sources for MIT license:
https://github.com/icons8/flat-color-icons/issues/5
https://github.com/icons8/flat-color-icons/blob/master/package.json

This covers plus_folder, bad_folder, chip, folder

Leaving plus, sd_card and star to need further research/replacement

The first two are user visible, the lock icons aren't used right now. 
![image](https://user-images.githubusercontent.com/190571/160492759-7e68b63e-0433-45df-8712-91376ddbb7bc.png)

The lock icons are shoddily replaced to avoid having to test and verify that removal doesn't break anything. 

Fixes #8077

Atemu commented on dist/license.md:
```diff
@@ -2,9 +2,9 @@ The icons in this folder and its subfolders have the following licenses:
-qt_themes/default/icons/16x16/checked.png | CC BY-ND 3.0 | https://icons8.com
-qt_themes/default/icons/16x16/failed.png | CC BY-ND 3.0 | https://icons8.com
-qt_themes/default/icons/16x16/lock.png | CC BY-ND 3.0 | https://icons8.com
+qt_themes/default/icons/16x16/checked.png |Apache 2.0 | https://github.com/google/material-design-icons/
+qt_themes/default/icons/16x16/failed.png | Apache 2.0 | https://github.com/google/material-design-icons/
+qt_themes/default/icons/16x16/lock.png | Apache 2.0 | https://github.com/google/material-design-icons/
```
Might not be a bad idea to mention that these were modified

Docteh replied:
Just slap `(modified)` after the url? or more details?

Tachi107 commented:
I find the checked.png icon hard to look at with a white background. Would it be possible to darken its green a bit?

Tachi107 commented on dist/license.md:
```diff
@@ -22,14 +22,14 @@ qt_themes/qdarkstyle/icons/48x48/folder.png | CC BY-ND 3.0 | https://icons8.com
-qt_themes/colorful/icons/48x48/folder.png | CC BY-ND 3.0 | https://icons8.com
+qt_themes/colorful/icons/256x256/plus_folder.png | MIT | https://github.com/icons8/flat-color-icons
+qt_themes/colorful/icons/48x48/bad_folder.png | MIT | https://github.com/icons8/flat-color-icons
+qt_themes/colorful/icons/48x48/chip.png | MIT | https://github.com/icons8/flat-color-icons
```
The `chip.png` shipped by yuzu is similar to the [`electronics.svg`](https://github.com/icons8/flat-color-icons/blob/8eccbbbd8b2af1d2c9593e7cfba5ecb0d68ee378/svg/electronics.svg) you can find in the flat-color-icons pack, but its not the same image

Tachi107 commented on dist/license.md:
```diff
@@ -22,14 +22,14 @@ qt_themes/qdarkstyle/icons/48x48/folder.png | CC BY-ND 3.0 | https://icons8.com
-qt_themes/colorful/icons/48x48/chip.png | CC BY-ND 3.0 | https://icons8.com
-qt_themes/colorful/icons/48x48/folder.png | CC BY-ND 3.0 | https://icons8.com
+qt_themes/colorful/icons/256x256/plus_folder.png | MIT | https://github.com/icons8/flat-color-icons
+qt_themes/colorful/icons/48x48/bad_folder.png | MIT | https://github.com/icons8/flat-color-icons
```
`bad_folder.png` and `plus_folder.png` are not from flat-color-icons, but are similar to [`folder.svg`](https://github.com/icons8/flat-color-icons/blob/8eccbbbd8b2af1d2c9593e7cfba5ecb0d68ee378/svg/folder.svg)

Docteh replied:
Yeah, modifications are open source, but its not clear if it'd stay MIT, or become GPLv2. 

Found a comment that mentions the modifier of the icons has signed yuzu CLA https://github.com/yuzu-emu/yuzu/pull/2898#issuecomment-538192851

Atemu replied:
I don't think including icon graphics in a GPL'd software triggers a GPL infection. It'd stay MIT unless the right holder of the modification decided to re-license under an MIT-compatible license such as the GPLv2.

IANAL.

Tachi107 commented on dist/license.md:
```diff
@@ -22,14 +22,14 @@ qt_themes/qdarkstyle/icons/48x48/folder.png | CC BY-ND 3.0 | https://icons8.com
 qt_themes/qdarkstyle/icons/48x48/sd_card.png | CC BY-ND 3.0 | https://icons8.com
 qt_themes/qdarkstyle/icons/48x48/star.png | CC BY-ND 3.0 | https://icons8.com
-qt_themes/colorful/icons/16x16/lock.png | CC BY-ND 3.0 | https://icons8.com
+qt_themes/colorful/icons/16x16/lock.png | Apache 2.0 | https://github.com/google/material-design-icons/
```
`lock.png` is already MIT licensed, see [`lock.svg`](https://github.com/icons8/flat-color-icons/blob/8eccbbbd8b2af1d2c9593e7cfba5ecb0d68ee378/svg/lock.svg)

Docteh commented:
I'm not actually resizing the images, its a crop
![image](https://user-images.githubusercontent.com/190571/160679527-8ff176ec-ed1a-44f5-8528-204e33fa5ffe.png)


Retint of the checked.png, I grabbed the color from the current checked.png
![image](https://user-images.githubusercontent.com/190571/160681680-abd0f441-6b2d-40c2-8b91-d3f7c685758d.png)

I'm going to convert this to a draft, just to avoid builds being generated when I push

Docteh commented:
Current icons
![image](https://user-images.githubusercontent.com/190571/160712148-69f9e717-41ab-4cea-be3c-079d2b12a322.png)


https://remixicon.com/ Apache 2.0 
![image](https://user-images.githubusercontent.com/190571/160712193-b79f70a8-2339-487e-b3da-4a75a333a24f.png)

Hero Icons, MIT https://github.com/tailwindlabs/heroicons

This one looks closer to what icons8 has. It doesn't have an SD Card, but that can be gotten to.
![image](https://user-images.githubusercontent.com/190571/160720098-15b00647-c7dc-4268-9c0c-449d01a4cf6b.png)

![bad_folder](https://user-images.githubusercontent.com/190571/160729404-469978ad-7abd-4336-ba3c-12c330d7de40.png)

Personally, I lean towards liking heroicons

@Tachi107 they have a LOT of icons to look through. 

- [ ] are heroicons good enough?
- [ ] sd cards

Docteh commented:
I've got all the ND icons replaced, so I squashed it down, and have double checked the changes to `dist/license.md`

Previous Colorful
![image](https://user-images.githubusercontent.com/190571/160865153-8fcc4c3b-0e2a-482f-b589-5dfd953c68d9.png)

sd_card, star, plus changed in this PR
![image](https://user-images.githubusercontent.com/190571/160865337-28af427b-a415-4fe3-af73-f96629bd85cf.png)


New White icons
![image](https://user-images.githubusercontent.com/190571/160864994-44fb272e-4ca5-4bef-a60c-4606000235a0.png)

Docteh commented:
Two changes, that I'll commit later. I don't want to spam builds. (These are the 48x48 pngs that will be commited later)

Star, Original color: EFCE4A different color: FFCA28 Set size to 42
![star](https://user-images.githubusercontent.com/190571/161444538-073a40cb-21ee-4bcb-ab00-b1818bca6a48.png)


SD Card is also a bit big, initially rasterized 48x48, better off at 44x44
![sd_card](https://user-images.githubusercontent.com/190571/161444536-7d068199-15d3-4e0b-b64a-f087499059ee.png)

Current PR/Preview with these changes
![image](https://user-images.githubusercontent.com/190571/161445348-c5e2245d-05ad-4071-bab0-916124beb3b9.png)![image](https://user-images.githubusercontent.com/190571/161445416-e00a4021-e2dd-4a17-8c11-82fb89eda1a9.png)

Docteh commented:
For the monochromatic icons I'm not sure if they should be filled in or not, but after staring at a lot of icons I believe they should be the same

Current
![image](https://user-images.githubusercontent.com/190571/162829800-e01df829-2c0b-4402-a6b9-0c4cae54e4ed.png)

This PR
![image](https://user-images.githubusercontent.com/190571/162829723-62a682f0-6398-45ee-a27e-23da8dada918.png)

IvarWithoutBones reviewed (approved):
We have been using a [rehosted patch](https://github.com/NixOS/nixpkgs/blob/e4d49de45a3b5dbcb881656b4e3986e666141ea9/pkgs/applications/emulators/yuzu/yuzu-free-icons.patch) of this PR for a few months now in [nixpkgs](https://github.com/NixOS/nixpkgs), as it allows us to redistribute binaries instead of having the end user compile it themselves. Redistributing the official binaries is not an option for us on NixOS, as our dynamic linker doesn't play nicely with that.

 It would be great if this could get merged!

Docteh commented:
#8541 adds 11 more icons, in that, 3 variants of 3, and 2 variants of 1 add up to 11. I complained over there about the issue, might need a reminder.

I got feedback from @german77 that the SD card looks like a foot, changed it out. Still a few pins short, but at 48x48 accuracy might he bard to see. 
![image](https://user-images.githubusercontent.com/190571/179763017-ccaf12ae-9228-480b-92aa-d8b8a9500659.png)

Redrew plus signs, the green felt too small, the others felt too big. I had a hard time seeing the exclamation mark in `failed`, so it's also replaced
![plus](https://user-images.githubusercontent.com/190571/179763188-702490cc-0282-4313-95f7-7c7ef67f8734.png) ![plus](https://user-images.githubusercontent.com/190571/179763238-fd46910b-4664-4197-a080-1c456add1370.png) ![plus](https://user-images.githubusercontent.com/190571/179763288-8584f4a9-f243-418a-8e3b-c08619cfd8f0.png) ![failed](https://user-images.githubusercontent.com/190571/179763857-20991dec-798d-4f7f-a1be-cc3c1d32d9a9.png)

Docteh commented:
@IvarWithoutBones Looks like #8541 got merged  https://github.com/yuzu-emu/yuzu/commit/705f7db84dd85555a6aef1e136cf251725cef293 so I'll need to dedicate some time for those.

IvarWithoutBones commented:
Thanks for the hard work and keeping us updated, its much appreciated ☺️

Docteh commented:
Big August Update!

... I reworked the changes to `dist/license.md` (now retired) to the new `.reuse/dep5` file


#8541 also adds some other icons, so

Remaining NO DERIVATIVES 
```
Files: dist/qt_themes/*/icons/16x16/connected.png
       dist/qt_themes/*/icons/16x16/connected_notification.png
       dist/qt_themes/*/icons/16x16/disconnected.png
       dist/qt_themes/*/icons/48x48/no_avatar.png
Copyright: https://icons8.com
License: CC-BY-ND-3.0
```

no_avatar candidates, https://ionic.io/ionicons https://github.com/ionic-team/ionicons (person)
![no_avatar](https://user-images.githubusercontent.com/190571/184495569-73ed02a7-deff-4cb2-ba4a-a6555d2b9824.png) ![no_avatar](https://user-images.githubusercontent.com/190571/184495577-41e164bb-481b-4e7c-8e2d-b31266ad52b6.png)

Docteh commented:
@Tachi107 I copy/pasted the Creative Commons Attribution-ShareAlike 3.0 Unported license into notepad. Seems okay. But maybe there is better source with files in .txt format?

Tachi107 commented:
> @Tachi107 I copy/pasted the Creative Commons Attribution-ShareAlike 3.0 Unported license into notepad. Seems okay. But maybe there is better source with files in .txt format?

Yes, there is. You can use `reuse download --all` to download all the licenses used by the project into the  'LICENSES/` folder.

You can install the `reuse` tool on Debian with `apt install reuse`, or via pip with `pip3 install reuse`

Docteh commented:
I swapped out the license with the .txt direct from Creative Commons. Why? easier to read on GitHub, or in less. 

Goes well with CC0-1.0.txt, also nicely formatted.

Tachi107 commented:
> I swapped out the license with the .txt direct from Creative Commons. Why? easier to read on GitHub, or in less.

Yeah I guess that is fine too. If the reuse CI job passes it means that everything is OK

Tachi107 commented:
Thanks for merging!
